src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ArgumentProvider.cs

```diff
@@ -0,0 +1,285 @@
+        {
+            var valueParam = new Parameter("value", null, _t, null, ValidationType.None, null);
+            var minParam = new Parameter("minimum", null, _t, null, ValidationType.None, null);
+            var maxParam = new Parameter("maximum", null, _t, null, ValidationType.None, null);
+            var argConstraints = new Dictionary<CSharpType, FormattableString>()
+            {
+                { _t, $"notnull, {typeof(IComparable<>)}" }
+            };
+            var signature = GetSignature("AssertInRange", new[] { valueParam, minParam, maxParam, _nameParam }, new[] { _t }, argConstraints);
+            var valueParamRef = new ParameterReference(valueParam);
+            return new Method(signature, new MethodBodyStatement[]
+            {
+                new IfStatement(GreaterThan(GetCompareToExpression(new ParameterReference(minParam), valueParamRef), Literal(0)))
+                {
+                    Throw(New.ArgumentOutOfRangeException(_nameParamRef, "Value is less than the minimum allowed.", false))
+                },
+                new IfStatement(LessThan(GetCompareToExpression(new ParameterReference(maxParam), valueParamRef), Literal(0)))
+                {
+                    Throw(New.ArgumentOutOfRangeException(_nameParamRef, "Value is greater than the maximum allowed.", false))
+                }
+            });
+        }
+
+        private ValueExpression GetCompareToExpression(ValueExpression left, ValueExpression right)
+        {
+            return left.Invoke("CompareTo", right);
+        }
+
+        private Method BuildAssertNotDefault()
+        {
+            var valueParam = new Parameter("value", null, _t, null, ValidationType.None, null);
+            var argConstraints = new Dictionary<CSharpType, FormattableString>()
+            {
+                { _t, $"struct, {typeof(IEquatable<>)}" }
+            };
+            var signature = GetSignature("AssertNotDefault", new[] { valueParam.WithRef(), _nameParam }, new[] { _t }, argConstraints);
+            return new Method(signature, new MethodBodyStatement[]
+            {
+                new IfStatement(new BoolExpression(new ParameterReference(valueParam).Invoke("Equals", Default)))
+                {
+                    ThrowArgumentException("Value cannot be empty.")
+                }
+            });
+        }
+
+        private Method BuildAssertNotNullOrWhiteSpace()
+        {
+            var valueParam = new Parameter("value", null, typeof(string), null, ValidationType.None, null);
+            var signature = GetSignature("AssertNotNullOrWhiteSpace", new[] { valueParam, _nameParam });
+            return new Method(signature, new MethodBodyStatement[]
+            {
+                AssertNotNullSnippet(valueParam),
+                new IfStatement(new BoolExpression(new InvokeStaticMethodExpression(typeof(string), "IsNullOrWhiteSpace", new[] { new ParameterReference(valueParam) })))
```

Comment:
I think we should add a `IsNullOrWhiteSpace` static method to `StringExpression` to introduce more readability here, and this method could return `BoolExpression` because `string.IsNullOrWhiteSpace` is returning `bool`.
```suggestion
                new IfStatement(StringExpression.IsNullOrWhiteSpace(value))
```